crates/sui-framework/packages/sui-framework/sources/event.move

```diff
@@ -37,6 +40,70 @@ module sui::event;
 /// phantom parameters, eg `emit(MyEvent<phantom T>)`.
 public native fun emit<T: copy + drop>(event: T);
 
+public use fun destroy_stream as EventStream.destroy;
+public use fun destroy_cap as EventStreamCap.destroy;
+public use fun emit_authenticated as EventStreamCap.emit;
+
+#[allow(unused_field)]
+public struct EventStreamHead has store {
+    /// Merkle root for all events in the current checkpoint.
+    root: vector<u8>,
+    /// Hash of the previous version of the head object.
+    prev: vector<u8>,
+}
+
+// A unique identifier for an event stream.
+public struct EventStream has key, store {
+    id: UID,
+}
+
+public fun new_event_stream(ctx: &mut TxContext): EventStream {
+    EventStream {
+        id: object::new(ctx),
+    }
+}
+
+public fun destroy_stream(stream: EventStream) {
+    let EventStream { id } = stream;
+    id.delete();
+}
+
+/// A capability to write to a given event stream
+public struct EventStreamCap has key, store {
+    id: UID,
+    stream_id: address,
+}
+
+public fun get_cap(stream: &EventStream, ctx: &mut TxContext): EventStreamCap {
+    EventStreamCap {
+        id: object::new(ctx),
+        stream_id: stream.id.to_address(),
+    }
+}
+
+public fun default_event_stream_cap<T: copy + drop>(ctx: &mut TxContext): EventStreamCap {
+    EventStreamCap {
+        id: object::new(ctx),
+        stream_id: type_name::get_original_package_id<T>(),
+    }
+}
+
+public fun destroy_cap(cap: EventStreamCap) {
+    let EventStreamCap { id, .. } = cap;
+    id.delete();
+}
+
+public fun emit_authenticated<T: copy + drop>(cap: &EventStreamCap, event: T) {
+    let accumulator_addr = accumulator::get_accumulator_field_address<EventStreamHead>(cap.stream_id);
+
+    emit_authenticated_impl<EventStreamHead, T>(accumulator_addr, cap.stream_id, event);
+}
+
+/// TODO: needs verifier rule like `emit` to ensure it is only called in package that defines `T`
```

Comment:
nit: the verifier rule should go on `emit_authenticated`, not `emit_authenticated_impl`.

Reply:
i think this was a stale comment, agreed